app/models/test.rb

```diff
@@ -1,7 +1,7 @@
 class Test < ApplicationRecord
   belongs_to :category
 
-  def self.sort_reverse(category)
-    where(category_id: Category.find_by(title: category).id).order('title DESC').pluck(:title)
+  def self.sort_reverse category
+    Test.joins("JOIN categories ON tests.category_id = categories.id and categories.title = '#{category}'").order(title: :desc).pluck(:title)
```

Comment:
имя класса можно не указывать, вызов идет у "селф"  и это собственно класс
```suggestion
    joins("JOIN categories ON tests.category_id = categories.id and categories.title = :category'", category: category).order(title: :desc).pluck(:title)
```
рекомендуется через именованные переменные делать
через знак вопроса планировали запретить в какой-то версии... может уже запретили
через интерполяцию делать не рекомендуется, это может быть потенциальной дырой в безопасности